spec/ruby/language/block_spec.rb

```diff
@@ -983,3 +983,77 @@ def a; 1; end
     end
   end
 end
+
+describe "Anonymous block forwarding" do
+  ruby_version_is "3.1" do
+    it "forwards blocks" do
+      eval <<-EOF
+          def b(&); c(&) end
+          def c(&); yield :non_null end
+      EOF
+
+      b { |c| c }.should == :non_null
+    end
+
+    it "requires the anonymous block parameter to be declared if directly passing a block" do
+      -> { eval "def a; b(&); end; def b; end" }.should raise_error(SyntaxError)
```

Comment:
It'd be good to check the message too here

Reply:
Okay. I think the current error message is not very good and I'd like to get it changed if possible. That'd break the spec, of course.

Reply:
I see, makes sense.
```
$ ruby -ve "def a; b(&); end; def b; end"
ruby 3.0.3p157 (2021-11-24 revision 3fb7d2cadc) [x86_64-linux]
-e:1: syntax error, unexpected ')'
def a; b(&); end; def b; end
-e:1: syntax error, unexpected end-of-input, expecting `end'

$ ruby -ve "def a; b(&); end; def b; end"
ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [x86_64-linux]
-e:1: no anonymous block parameter
```
So this no longer a SyntaxError in 3.1, so we need a version guard, or remove this spec.

Reply:
Actually that's also a SyntaxError on 3.1, weird that CRuby doesn't show that.
```
$ ruby -ve 'eval "def a; b(&); end; def b; end" rescue p $!'
ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [x86_64-linux]
-e:1:in `eval': (eval):1: no anonymous block parameter (SyntaxError)
	from -e:1:in `<main>'
```
So the spec is fine as-is.